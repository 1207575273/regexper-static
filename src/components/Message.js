import React from 'react';
import PropTypes from 'prop-types';

const renderIcon = icon => {
  if (!icon) {
    return;
  }

  const Icon = icon;
  return <Icon/>;
};

const Message = ({ icon, heading, children }) => (
  <div className="message">
    <h2>{ renderIcon(icon) }{ heading }</h2>
    { children }
  </div>
);

Message.propTypes = {
  icon: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func
  ]),
  heading: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default Message;
