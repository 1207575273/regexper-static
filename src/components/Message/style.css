.message {
  background: var(--color-tan);
  padding: var(--spacing-margin);
  margin: var(--spacing-margin) 0;
}

.message h2 {
  margin: var(--spacing-margin) 0;
}

.message h2 svg {
  margin-right: 0.5rem;
  height: 3rem;
  width: 3rem;
  vertical-align: bottom;
}

.message p {
  margin: var(--spacing-margin) 0;
}
