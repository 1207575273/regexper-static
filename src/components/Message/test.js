import React from 'react';
import { shallow, render } from 'enzyme';

import Message from './index';

test('Message rendering', () => {
  const component = shallow(
    <Message heading="Testing" className="testing">
      <p>Message content</p>
    </Message>
  );
  expect(component).toMatchSnapshot();
});

test('Message rendering with icon', () => {
  const Icon = () => 'Sample icon SVG';
  const component = render(
    <Message heading="Testing" icon={ Icon }>
      <p>Message content</p>
    </Message>
  );
  expect(component).toMatchSnapshot();
});
