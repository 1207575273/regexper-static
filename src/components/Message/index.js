import React from 'react';
import PropTypes from 'prop-types';

import style from './style.css';

import InfoIcon from 'feather-icons/dist/icons/info.svg';
import ErrorIcon from 'feather-icons/dist/icons/alert-octagon.svg';
import WarningIcon from 'feather-icons/dist/icons/alert-triangle.svg';

const iconTypes = {
  info: InfoIcon,
  error: ErrorIcon,
  warning: WarningIcon
};

const renderIcon = (type, icon) => {
  icon = icon || iconTypes[type];

  if (!icon) {
    return;
  }

  const Icon = icon;
  return <Icon/>;
};

const Message = ({ type, icon, heading, children }) => (
  <div className={ [ style.message, type && style[type] ].filter(Boolean).join(' ') }>
    <div className={ style.header }>
      { renderIcon(type, icon) }
      <h2>{ heading }</h2>
    </div>
    <div className={ style.content }>
      { children }
    </div>
  </div>
);

Message.propTypes = {
  type: PropTypes.oneOf([
    'info',
    'error',
    'warning'
  ]),
  icon: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func
  ]),
  heading: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default Message;
