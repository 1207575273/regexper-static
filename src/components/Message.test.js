import React from 'react';
import renderer from 'react-test-renderer';

import Message from './Message';

test('Message rendering', () => {
  const component = renderer.create(
    <Message heading="Testing">
      <p>Message content</p>
    </Message>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Message rendering with icon', () => {
  const Icon = () => 'Sample icon SVG';
  const component = renderer.create(
    <Message heading="Testing" icon={ Icon }>
      <p>Message content</p>
    </Message>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
