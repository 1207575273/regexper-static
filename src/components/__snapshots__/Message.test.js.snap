// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message rendering 1`] = `
<div
  className="message"
>
  <h2>
    Testing
  </h2>
  <p>
    Message content
  </p>
</div>
`;

exports[`Message rendering with icon 1`] = `
<div
  class="message"
>
  <h2>
    Sample icon SVGTesting
  </h2>
  <p>
    Message content
  </p>
</div>
`;
